### ISSUE TITLE
think about upgrading proofs

### ISSUE THREAD

**Issue by whyrusleeping** [filecoin-project/specs#132]
We should consider how PoSts will work if we have to upgrade how (for example) Seals work. Will miners have to re-seal all their data? Will we have to run two PoSts in parallel for a while? will we do a 'soft upgrade' that just doesnt allow adding new seals of the old kind? 

The action item for this issue isnt necessarily to solve all the problems, but we need to brainstorm potential situations and plan for them.

**Comment by pooja** [comment]
@porcuquine Can you help provide some thoughts here? Thanks!

**Comment by porcuquine** [comment]
My initial thoughts:

We should continue to support already-sealed sectors for as long as necessary (until they expire). That would mean having a versioned PoSt and would imply that we need to be able to know what version of Seal was used. I think we can keep this simple by only ever supporting one version of Seal. That would mean we know what kind of PoSt a sector needs based on when it was sealed (that will never change).

If a miner doesn't like having to run two PoSts, they can reseal. This doesn't require any extra support.

I don't think I'm introducing anything new, just answering @whyrusleeping's questions and expressing a preference for a particular strategy.

**Comment by whyrusleeping** [comment]
> they can reseal.

We should document how resealing a live deal would work. The trickiest bits are:
- preventing the client from slashing them for no longer having the old sector
- making the payment still valid for the new sector

(can probably make a new issue to discuss that)
